src/client_side_request.h

```diff
@@ -72,14 +72,31 @@ class ClientHttpRequest
         }
     }
 
+    /// Initializes the current request with the virgin request.
+    /// Call this method when the virgin request becomes known.
+    /// To update the current request later, use resetRequest().
+    void initRequest(HttpRequest *);
+
+    /// Resets the current request to the latest adapted or redirected
+    /// request. Call this every time adaptation or redirection changes
+    /// the request. To set the virgin request, use initRequest().
+    void resetRequest(HttpRequest *);
+
     /** Details of the client socket which produced us.
      * Treat as read-only for the lifetime of this HTTP request.
      */
     Comm::ConnectionPointer clientConnection;
 
-    HttpRequest *request;       /* Parsed URL ... */
+    /// Request currently being handled by ClientHttpRequest.
+    /// Starts as a virgin request; see initRequest().
+    /// Adaptation and redirections replace it; see resetRequest().
+    HttpRequest * const request;
     char *uri;
```

Comment:
Having worked so much with the URI-related code, you probably know what this `uri` member is. Let's try to document it, exposing the difference between `uri`, `log_uri`, and (indirectly) `request->url`.